src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -2548,6 +2548,15 @@ public function seek( $bookmark_name ): bool {
 			return false;
 		}
 
+		$existing_bookmark = $this->bookmarks[ $bookmark_name ];
+
+		if (
+			$this->token_starts_at === $existing_bookmark->start &&
+			$this->token_length === $existing_bookmark->length
+		) {
+			return true;
+		}
```

Comment:
How will this impact (or not) `WP_HTML_Processor::seek()` since its actual bookmarks get prefixed with `_`? 

https://github.com/WordPress/wordpress-develop/blob/b071c28f8b6be06e464ccf582877e66c02859679/src/wp-includes/html-api/class-wp-html-processor.php#L5266-L5275

Reply:
it shouldn't because by the time it calls this parent-class method, it has supplied the underscore.

Reply:
OK, right. By the way, what is the reasoning behind prefixing bookmarks with underscores in HTML Processor? In https://github.com/WordPress/performance/pull/1546 I'm making use of `WP_HTML_Processor` when available, but in the logic that injects additional markup into the `head` and `body`, it needs to obtain the `start` position of the bookmark. For the HTML Processor version it has to prepend the `_` which seems strange, although granted this is an internal API:

* [`WP_HTML_Tag_Processor` version](https://github.com/WordPress/performance/blob/430f75d6669248b81973f1ef770379c8d69108e6/plugins/optimization-detective/class-od-html-tag-processor.php#L571-L600)
* [`WP_HTML_Processor` version](https://github.com/WordPress/performance/blob/430f75d6669248b81973f1ef770379c8d69108e6/plugins/optimization-detective/class-od-html-processor.php#L359-L391)

Reply:
the reason was to avoid any possible conflicts between bookmarks set by the HTML Processor itself, and the user-space code setting bookmarks. that is, any time someone calls `WP_HTML_Processor::set_bookmark()` it will be prefixed, but calls _inside_ the processor to `parent::set_bookmark()` will be un-prefixed. it's a very basic namespacing mechanism.